contracts/AcceleratingDistributor.sol

```diff
@@ -156,30 +154,36 @@ contract AcceleratingDistributor is ReentrancyGuard, Ownable, Multicall {
         // payouts could eat into user balances. maxMultiplier is constrained to be at least 1e18 to enforce a minimum
         // 1x multiplier and avoid potential underflows.
         require(stakedToken != address(rewardToken), "Staked token is reward token");
+        require(baseEmissionRate <= 1e24, "baseEmissionRate too large"); // 1 million tokens per second.
         require(maxMultiplier <= 1e24, "maxMultiplier too large"); // 1_000_000x multiplier.
         require(maxMultiplier >= 1e18, "maxMultiplier less than 1e18");
         require(secondsToMaxMultiplier > 0, "secondsToMaxMultiplier is 0");
-        require(baseEmissionRate <= 1e24, "baseEmissionRate too large"); // 1 million tokens per second.
 
         StakingToken storage stakingToken = stakingTokens[stakedToken];
+        require(stakingToken.lastUpdateTime == 0, "Staking token already added");
 
-        // If this token is already initialized, make sure we update the rewards before modifying any params.
-        if (stakingToken.lastUpdateTime != 0) _updateReward(stakedToken, address(0));
-
-        stakingToken.enabled = enabled;
         stakingToken.baseEmissionRate = baseEmissionRate;
         stakingToken.maxMultiplier = maxMultiplier;
         stakingToken.secondsToMaxMultiplier = secondsToMaxMultiplier;
-        stakingToken.lastUpdateTime = getCurrentTime();
 
-        emit TokenConfiguredForStaking(
-            stakedToken,
-            enabled,
-            baseEmissionRate,
-            maxMultiplier,
-            secondsToMaxMultiplier,
-            stakingToken.lastUpdateTime
-        );
+        emit TokenAddedForStaking(stakedToken, baseEmissionRate, maxMultiplier, secondsToMaxMultiplier);
+    }
+
+    /**
+     * @notice Change the configuration of a staking token. Emits a StakingTokenUpdated event on success.
+     * @param stakedToken The address of the token that can be staked.
+     * @param enabled Whether the token is enabled for staking.
+     */
+    function configureStakingToken(address stakedToken, bool enabled) external onlyOwner {
+        require(stakedToken != address(0), "Invalid staking token");
+
+        StakingToken storage stakingToken = stakingTokens[stakedToken];
+        require(stakingToken.maxMultiplier >= 1e18, "Staking token not added");
```

Comment:
I don't think you need to check the `stakingToken` params here since we can assume that if it was added via `addStakingToken` then its params were set correctly. This probably requires that `lastUpdateTime > 0` to ensure that the staking token was added

Reply:
That's actually what this check was intending to do. Checking the timestamp is a more obvious way of doing it, but at the moment the PR doesn't propose to update the timestamp in addStakingToken(), because that permits the token configuration to be modified multiple times before it is initially enabled. It's debatable as to whether that's useful or not.

In any case, it's probably a moot point since we're like to just document the implications of modifying rewards.